src/linker/Linker/Annotations.cs

```diff
@@ -561,14 +561,17 @@ public bool SetPreservedStaticCtor (TypeDefinition type)
 		/// <returns>Returns true along with the RequiresUnreferencedCodeAttribute if found, otherwise returns false</returns>
 		public bool TryGetEffectiveRequiresUnreferencedCodeAttributeOnType (TypeDefinition type, out RequiresUnreferencedCodeAttribute attribute)
 		{
+			attribute = null;
+			if (type == null)
+				return false;
+
 			do {
 				if (TryGetLinkerAttribute (type, out RequiresUnreferencedCodeAttribute declaringTypeAttribute)) {
 					attribute = declaringTypeAttribute;
 					return true;
 				}
 				type = type.DeclaringType;
 			} while (type != null);
```

Comment:
What about changing do-while to `while (type != null)` instead?

Reply:
Fixed